src/server/pachyderm_test.go

```diff
@@ -11641,4 +11641,23 @@ func TestDatumBatching(t *testing.T) {
 		request := createPipelineRequest(pipeline, script)
 		checkState(request, pps.JobState_JOB_FAILURE)
 	})
+	t.Run("Env", func(t *testing.T) {
+		// This test will error if the datum id environment variable isn't set and distinct for each datum.
+		script := fmt.Sprintf(`
+			set -a
```

Comment:
Not a required change; but I like to start shell scripts with "set -euo pipefail" just to find unexpected weirdness as quickly as possible.  (-e makes the script exit as soon as there is a non-zero command, -u makes unset variable references fatal, pipefail makes pipelines return the "worst" exit status in the pipeline; no pipes here so not relevant).  I don't actually know what "a" does (I guess it makes "source" create readable variables even though the text of the file is FOO=bar and not export FOO=bar?)

Reply:
yeah, it exports created variables and functions.